pvlib/tests/test_irradiance.py

```diff
@@ -1129,10 +1130,30 @@ def test_dni():
     ghi = pd.Series([90, 100, 100, 100, 100])
     dhi = pd.Series([100, 90, 50, 50, 50])
     zenith = pd.Series([80, 100, 85, 70, 85])
-    clearsky_dni = pd.Series([50, 50, 200, 50, 300])
+    dni_clear = pd.Series([50, 50, 200, 50, 300])
 
     dni = irradiance.dni(ghi, dhi, zenith,
-                         clearsky_dni=clearsky_dni, clearsky_tolerance=2)
+                         dni_clear=dni_clear, clearsky_tolerance=2)
+    assert_series_equal(dni,
+                        pd.Series([float('nan'), float('nan'), 400,
+                                   146.190220008, 573.685662283]))
+
+    dni = irradiance.dni(ghi, dhi, zenith)
+    assert_series_equal(dni,
+                        pd.Series([float('nan'), float('nan'), 573.685662283,
+                                   146.190220008, 573.685662283]))
+
+
+@fail_on_pvlib_version("0.12")
+def test_dni_dni_clearsky_deprecation():
```

Comment:
We should add a similar `with pytest.warns` check for the `dirindex` function.  Can be in this same test function or a new one, up to you.